### PR TITLE
 Display graphical statistics on transactions page - Closes #2675, #2676, #2677, #2678

### LINKED ISSUE #2470: Display transaction statistics
### Description
As a user; I'd like to see a summary of transaction in test and graphs format which gives me overall understanding about information such as:
- The proportion of each transaction type
- The distribution of amount ranges per transaction
- Volumes
- Number of transactions

### Acceptance criteria
According to the [design guides](https://projects.invisionapp.com/d/main#/console/18539694/389393302/preview):

### Tickets
- Display number of transaction statistics #2675
- Display volume of transaction statistics #2676
- Display amount per transaction statistics #2677
- Display types of transaction statistics #2678

### LINKED ISSUE #2675: Display number of transaction statistics
### Description

Display statistics for number of transaction occurred in last 7 days, 6 months and 12 months

### Motivation

Provide users with realtime stats on number of transaction throughput

### Acceptance Criteria
1. Number of transaction occurred in last 7 days from the current date & time.
2. Number of transaction occurred in last 6 months from the current date & time.
3. Number of transaction occurred in last 12 months from the current date & time.
4. The data should support to build bar graph
5. All the transaction details should be in UTC time zone.

### Additional Information
Refer [UI design](https://projects.invisionapp.com/share/ZRU30GY5NTX#/screens/389319981) for better information.

Screenshot 2019-11-25 at 11 14 09

### LINKED ISSUE #2678: Display types of transaction statistics
### Description
Display statistics for types of transaction occurred in last 7 days, 6 months and 12 months

### Motivation
Provide users with realtime stats on types of transaction throughput

### Acceptance Criteria

1. Types of transaction occurred in last 7 days from the current date & time.
2. Types of transaction occurred in last 6 months from the current date & time.
3. Types of transaction occurred in last 12 months from the current date & time.
4. We should support 8 types of transaction as per the current system Architecture and this should be scalable enough to support new transaction types in future.
5. The data should support to build donut graph and pie graph
6. All the transaction details should be in UTC time zone

### Additional Information
Refer [UI design](https://projects.invisionapp.com/share/ZRU30GY5NTX#/screens/389319981) for better information.

Screenshot 2019-11-25 at 11 13 50

### PR DESCRIPTION
**Pull request by reyraa**
### What was the problem?
This PR resolves #2470

### How was it solved?
- Export charts by their types.
- Add overview section to transactions
- Fetch data and normalize (wherever required)
- Display data using charts

### CODE REVIEW

**Comment by slaweet** [comment]
> Transaction statistic charts don't work on Testnet and Mainnet.

That's expected. The API is coming in the Lisk Service 1.2.0 that should go to production early next week (as discussed in the standup today).

File: src/components/toolbox/charts/index.js [hunk 1]
```diff
@@ -2,39 +2,85 @@ import React from 'react';
...
-export default Chart;
+LineChart.propTypes = propTypes;
+BarChart.propTypes = propTypes;
+DoughnutChart.propTypes = propTypes;
+
+export default {
+  LineChart,
+  BarChart,
+  DoughnutChart,
+};
```
Comment (slaweet) [inline src/components/toolbox/charts/index.js]: Nice refactor 👍

File: src/components/toolbox/charts/index.js [hunk 2]
```diff
@@ -2,39 +2,85 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { useTheme } from '../../../utils/theme';
 import { CHART_TYPES } from '../../../constants/chartConstants';
```
Comment (slaweet) [inline src/components/toolbox/charts/index.js]: With your refactor, I think that `CHART_TYPES` is not needed anymore. You can directly
```js
import { Line, Bar, Doughnut } from 'react-chartjs-2';
```

File: src/components/screens/monitor/transactions/overview.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                    label: 'ABC',
+                    data: Object.values(distributionByAmount),
+                  },
+                ],
+              }}
+            />
+          </div>
+        </div>
+        <div className={`${styles.column} ${styles.bar}`}>
+          <h2 className={styles.title}>{t('No transactions / volume (LSK)')}</h2>
```
Comment (slaweet) [inline src/components/screens/monitor/transactions/overview.js]:
```suggestion
          <h2 className={styles.title}>{t('No of transactions / volume (LSK)')}</h2>
```

File: src/components/screens/monitor/transactions/overview.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        </div>
+        <div className={`${styles.column} ${styles.pie}`}>
+          <h2 className={styles.title}>{t('Amount per transaction (LSK)')}</h2>
+          <div className={styles.graph}>
+            <DoughnutChart
+              data={{
+                labels: Object.keys(distributionByAmount),
+                datasets: [
+                  {
+                    label: 'ABC',
```
Comment (slaweet) [inline src/components/screens/monitor/transactions/overview.js]: is the `label` mandatory here?

**Comment by slaweet** [comment]
And I found the following functional issues
1. Overlap on smaller screen
2. Color too similar to "Delegate registration
3. Labels not transformed (e.g. 1_10 -> 1 < 10)
4. Labels not transformed, e.g. 8000 -> 8k
5. Labels not transformed, e.g. 03-06 -> Friday
6. Axis line missing
7. Might be worth to merge
8. Reverse order

Screenshot 2020-03-06 at 10 12 58
Screenshot 2020-03-06 at 10 12 20

**Comment by reyraa** [comment]
@slaweet @yasharAyari 
Thank you guys. I fixed most of the issues you spotted. remaining: 
 - Axis line missing: I can't seem to find an solution that removes the grids but keeps the axis. they appear and disappear together. I appreciate your help here.
- Might be worth to merge these two: When I suggested this (did I?) I thought the values under 1LSK are negligible. but they're not. I rather keep it. but what do you think? now if we keep it, @slaweet why the sequence of the response is wrong? can we sort it on Service?

**Comment by slaweet** [comment]
Thank you for the changes, @reyraa . I checked everything again.

> why the sequence of the response is wrong? 

if you mean `distributionByAmount`, then I think we discussed this in some meeting - in the API it's an object you cannot assume it's ordered.

I realized another issue I missed before:
- 🐛 The transaction types seem wrong on betanet (No "send" transaction?)
- 🐛 "Number" and "Volume" seem to be swapped - I expect values of Volume to be higher
- 🎨 The green color in "Number" label looks darker than the color in the chart 

Now I tested it also with mainnet and testnet:
Screenshot 2020-03-06 at 15 29 24
Screenshot 2020-03-06 at 15 32 36

- 🎨 There are more amount ranges than colors. Looking at the data, I think all ranges above 10K can be merged.

**Comment by slaweet** [comment]
> * Axis line missing: I can't seem to find an solution that removes the grids but keeps the axis. they appear and disappear together. I appreciate your help here.

https://www.chartjs.org/samples/latest/scales/gridlines-display.html demonstrates it's possible. Source of that page should show how - something with drawBorder: true and drawOnChartArea: false.

**Comment by slaweet** [comment]
Screenshot 2020-03-06 at 16 24 00

I realized some more design discrepancies:
- 🎨 "year" tab doesn't have rounded corner when selected
- 🎨 All tabs should have a min-width
- 🎨 Font-size of "Number" and "Volume" is bigger than in the design

File: src/components/screens/monitor/transactions/overview.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+};
+
+const tabs = (t = str => str) => [
+  {
+    value: 'week',
+    name: t('{{num}} week', { num: 1 }),
+  },
+  {
+    value: 'month',
+    name: t('{{num}} month', { num: 6 }),
```
Comment (slaweet) [inline src/components/screens/monitor/transactions/overview.js]: Typo:
```suggestion
    name: t('{{num}} months', { num: 6 }),
```

File: src/components/screens/monitor/transactions/overview.js [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+              <h5 className={`${styles.legend} ${styles.number}`}>
+                <span>{t('Number')}</span>
+                <Tooltip className={`${styles.tooltip} showOnLeft`}>
+                  <p>{t('The number of transactions submitted on the given time period.')}</p>
+                </Tooltip>
+              </h5>
+              <h5 className={`${styles.legend} ${styles.volume}`}>
+                <span>{t('Volume')}</span>
+                <Tooltip className={`${styles.tooltip} showOnLeft`}>
+                  <p>{t('The aggregated of LSK volume transferred on the given time period.')}</p>
```
Comment (slaweet) [inline src/components/screens/monitor/transactions/overview.js]: I'm not sure where is this copy from but I like it more than the one in [design](https://projects.invisionapp.com/d/main#/console/18539694/390074223/preview). I just think that to be grammatically correct, it should say:

```suggestion
                  <p>{t('The aggregate of LSK volume transferred on the given time period.')}</p>
```

File: src/components/screens/monitor/transactions/overview.js [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+  }
+  return date.replace(/^\d{4}-/, '');
+};
+
+const formatDistributionBValues = distributions => [0, 1, 2, 3, 4].map(item =>
+  (distributions[item] || 0) + (distributions[item + 8] || 0));
+
+const Overview = ({ t, txStats }) => {
+  const [activeTab, setActiveTab] = useState('week');
+  const distributionByType = formatDistributionBValues(txStats.data.distributionByType);
```
Comment (slaweet) [inline src/components/screens/monitor/transactions/overview.js]: Typo:
```suggestion
const formatDistributionByValues = distributions => [0, 1, 2, 3, 4].map(item =>
  (distributions[item] || 0) + (distributions[item + 8] || 0));

const Overview = ({ t, txStats }) => {
  const [activeTab, setActiveTab] = useState('week');
  const distributionByType = formatDistributionByValues(txStats.data.distributionByType);
```

**Comment by reyraa** [comment]
@slaweet I resolved the issues. I hope this satisfies the changes you requested.

File: src/components/screens/monitor/transactions/overview.js (hunk 5 above)
Comment (slaweet) [reply src/components/screens/monitor/transactions/overview.js]: @reyraa the tab still says "6 month" instead of "6 months"

Comment (slaweet) [reply src/components/screens/monitor/transactions/overview.js]: Also, I still don't see the `min-width` of tabs I mentioned before. In designs it's 128px

**Comment by slaweet** [comment]
Thank you, Ali. Now it looks pretty good. Just two tiny things left: https://github.com/LiskHQ/lisk-desktop/pull/2813#discussion_r389716599

File: src/components/screens/monitor/transactions/overview.js (hunk 5 above)
Comment (reyraa) [reply src/components/screens/monitor/transactions/overview.js]: @slaweet I didn't add min-width deliberately to match the other pages.

**Pull request merged**